front/LimitedEntryMap.js

```diff
@@ -42,6 +42,11 @@ module.exports = function(maxEntries){
         },
         keys: function(){
             return Object.keys(map);
+        },
+        remove: function(key){
+            if (keys.indexOf(key) !== -1)
+                delete map[key];
```

Comment:
This should remove `key` from `keys` too.
See http://stackoverflow.com/a/5767357